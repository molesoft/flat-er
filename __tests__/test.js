import flat from '../src/index'

it('Should flatten', () => {
  const obj = {
    foo: 'bar',
    baz: 2,
    dogs: [
      "lab",
      "great dane"
    ],
    cat: {
      whiskers: true,
      litterBox: {
        clean: false
      },
      colors: [
        'black',
        'white'
      ]
    }
  }
  const flattened = flat.flatten(obj)
  expect(flattened).toMatchSnapshot()
})