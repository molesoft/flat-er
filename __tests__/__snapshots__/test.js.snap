// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should flatten 1`] = `
Object {
  "baz": 2,
  "cat.colors.0": "black",
  "cat.colors.1": "white",
  "cat.litterBox.clean": false,
  "cat.whiskers": true,
  "dogs.0": "lab",
  "dogs.1": "great dane",
  "foo": "bar",
}
`;
